client/landing/gutenboarding/onboarding-block/create-site/create-and-redirect.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { Redirect } from 'react-router-dom';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
+import { USER_STORE } from '../../stores/user';
+import { useFreeDomainSuggestion } from '../../hooks/use-free-domain-suggestion';
+import { Step, usePath } from '../../path';
+
+type Status = 'INIT' | 'SUCCESS' | 'ERROR' | 'MISSING_SITE_DATA';
+
+const CreateAndRedirect = () => {
+	const { siteTitle, siteVertical } = useSelect( select => select( ONBOARD_STORE ).getState() );
+	const [ status, setStatus ] = useState< Status >( siteVertical ? 'INIT' : 'MISSING_SITE_DATA' );
+
+	const currentUser = useSelect( select => select( USER_STORE ).getCurrentUser() );
+	const makePath = usePath();
+	const { createSite } = useDispatch( ONBOARD_STORE );
+
+	const freeDomainSuggestion = useFreeDomainSuggestion();
```

Comment:
hmm this always came as undefined for me. I tested with no site title and with a site title like `cars.com`. with `cars.com` it doesn't get called, with no site title it is always undefined

Reply:
Returning `undefined` is actually expected if there's no site title. If they haven't entered a site title then we don't suggest anything (notice there's no domain picker in the header either).

But `undefined` when the site title is `cars.com` is a bit strange 😕 It will return `undefined` initially while loading but eventually should return a suggestion. When I tried `cars.com` it returned a suggestion of `carsonlineshop.car.blog`. I'll keep digging and see if I can figure out what's happening for you.

Reply:
ahh yep, I retested and saw the same behaviour as you. hmm I'm not sure maybe I just missed it the first time.